Centaurus.Domain/Transactions/Parsers/PaymentsParserBase.cs

```diff
@@ -0,0 +1,52 @@
+﻿using Centaurus.Models;
+using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace Centaurus.Domain
+{
+    public abstract class PaymentsParserBase
+    {
+        public abstract PaymentProvider Provider { get; }
+
+        public abstract bool TryDeserializeTransaction(byte[] rawTransaction, out TransactionWrapper transaction);
+
+        public abstract void ValidateTransaction(TransactionWrapper transaction, string vault);
+        
+        public abstract WithdrawalWrapper GetWithdrawal(MessageEnvelope envelope, TransactionWrapper transactionWrapper, ConstellationSettings constellationSettings, string vault);
+
+        public abstract int CompareCursors(string left, string right);
+
+        public abstract object ParseSecret(string secret);
```

Comment:
There's no need to expose public `ParseSecret`, `ParseVault`, `ParseCursor`, `CompareCursors` - the logic should be encapsulated inside the provider class. You can expose two methods - `LoadSettings` which takes the config, finds the corresponding settings section, and parses all variables - secret, vault, cursor, etc, and `ProcessPayment` method responsible for the payment handling logic (encapsulates `CompareCursors` and other functionality).

This way we'll have more flexible logic.